x/group/spec/messages.md

```diff
@@ -0,0 +1,46 @@
+```go
+// Groups get their own GroupID
+type MsgCreateGroup struct {
+	Signer sdk.AccAddress `json:"signer"`
+	// The Owner of the group is allowed to change the group structure. A group account
+	// can own a group in order for the group to be able to manage its own members
+	Owner  sdk.AccAddress `json:"owner"`
+	// The members of the group and their associated weight
+	Members []Member `json:"members,omitempty"`
+	// TODO maybe make this something more specific to a domain name or a claim on identity? or Info leave it generic
+	Memo string `json:"memo,omitempty"`
+}
+
+// group accounts get their own sdk.AccAddress
+type MsgCreateGroupAccount struct {
+	Signer         sdk.AccAddress `json:"signer"`
+	// The Owner of a group account is allowed to change the DecisionPolicy. This can be left nil 
+	// in order for the group account to "own" itself
+	Owner          sdk.AccAddress `json:"owner"`
+	Group          GroupID        `json:"group"`
+	DecisionPolicy DecisionPolicy `json:"decision_policy"`
+	Memo           string         `json:"memo,omitempty"`
+}
+type MsgCreateProposal struct {
+	Proposer sdk.AccAddress `json:"proposer"`
+	GroupAccount    sdk.AccAddress `json:"group"`
+	Msgs     []sdk.Msg      `json:"msgs"`
+	// Exec can be set to true in order to attempt to execute the proposal immediately
+	// with no voting in a single transaction - this is useful for 1/N or 2/N multisig
+	// key groups. Every signer of the MsgCreateProposal transaction is considered a yes
```

Comment:
There is a problem here, we don't have access to the signers list in the Handler.
(I have an open issue on this one: https://github.com/cosmos/cosmos-sdk/issues/4937 )

Currently, the work-around is to define all signers you want to verify in the msg itself, and return them in `GetSigners()`. So... we would need to add another array of "co-signers" here to allow 2 of N as positive votes.

Reply:
Also, in general, can we assume that the proposal also makes a "yes" vote upon "create proposal", even if exec is false?